tempesta_fw/t/functional/testers/functional.py

```diff
@@ -55,13 +55,24 @@ def setUp(self):
 
     def tearDown(self):
         # Close client connection before stopping the TempestaFW.
-        asyncore.close_all()
-        if self.client:
-            self.client.close()
-        if self.tempesta:
-            self.tempesta.stop()
-        if self.tester:
-            self.tester.close_all()
+        try:
+            asyncore.close_all()
+            if self.client:
+                self.client.close()
+        except:
```

Comment:
Seems that `client` class must be stopped just in the same way as `Nginx` and `Tempesta` classes. So it's also statefull.